opensrp-path/src/main/java/org/ei/opensrp/path/tabfragments/ChildUnderFiveFragment.java

```diff
@@ -145,11 +144,11 @@ private void createWeightLayout(LinearLayout fragmentContainer, boolean editmode
                 }
             }
             if (!formattedAge.equalsIgnoreCase("0d")) {
-                weightmap.put(formattedAge, weightlist.get(i).getKg() + " kg");
+                weightmap.put(weight.getId(), Pair.create(formattedAge, weightlist.get(i).getKg() + " kg"));
```

Comment:
this and line 187 feel like they are begging for a `kilogramStringSuffix` helper or similar